extensions/ql-vscode/src/remote-queries/export-results.ts

```diff
@@ -1,41 +1,50 @@
 import * as path from 'path';
 import * as fs from 'fs-extra';
 
-import { window, commands, Uri, ExtensionContext, QuickPickItem, workspace, ViewColumn } from 'vscode';
+import { window, commands, Uri, ExtensionContext, workspace, ViewColumn } from 'vscode';
 import { Credentials } from '../authentication';
 import { UserCancellationException } from '../commandRunner';
 import { showInformationMessageWithAction } from '../helpers';
 import { logger } from '../logging';
 import { QueryHistoryManager } from '../query-history';
 import { createGist } from './gh-api/gh-api-client';
 import { RemoteQueriesManager } from './remote-queries-manager';
-import { generateMarkdown } from './remote-queries-markdown-generation';
+import {
+  generateMarkdown,
+  generateVariantAnalysisMarkdown,
+  MarkdownFile,
+} from './remote-queries-markdown-generation';
 import { RemoteQuery } from './remote-query';
 import { AnalysisResults, sumAnalysesResults } from './shared/analysis-result';
 import { pluralize } from '../pure/word';
+import { VariantAnalysisManager } from './variant-analysis-manager';
+import { assertNever } from '../pure/helpers-pure';
+import {
+  VariantAnalysis,
+  VariantAnalysisScannedRepository,
+  VariantAnalysisScannedRepositoryResult
+} from './shared/variant-analysis';
 
 /**
- * Exports the results of the currently-selected remote query.
+ * Exports the results of the currently-selected remote query or variant analysis.
  */
 export async function exportSelectedRemoteQueryResults(queryHistoryManager: QueryHistoryManager): Promise<void> {
   const queryHistoryItem = queryHistoryManager.getCurrentQueryHistoryItem();
   if (!queryHistoryItem || queryHistoryItem.t === 'local') {
     throw new Error('No variant analysis results currently open. To open results, click an item in the query history view.');
   }
 
-  if (!queryHistoryItem.completed) {
```

Comment:
Don't we still want to perform this check for remote queries?

Reply:
Yes, we do still need it for the remote queries. However, I felt it was redundant to do it twice, so this is now only checked in [the command we call from this method](https://github.com/github/vscode-codeql/blob/e9681bc5468d846a1549f7e4abb1afbdd5f3a649/extensions/ql-vscode/src/remote-queries/export-results.ts#L62-L64). This should result in the same behaviour for the user.

Reply:
Ah I see!